app/models/project_step.rb

```diff
@@ -42,6 +42,22 @@ class ProjectStep < ActiveRecord::Base
   validates :project_id, presence: true
 
 
+  def division
+    project.try(:division)
+  end
+
+  def update_with_translations(project_step_params, translations_params)
+    begin
+    ActiveRecord::Base.transaction do
+      update_translations!(translations_params)
+      update!(project_step_params)
```

Comment:
Not sure about this approach. Could perhaps just use nested attributes? See comment below in file.